### ISSUE TITLE
huge PR - skiasharp support

### ISSUE THREAD

**Issue by gentledepp** [vvvv/SVG#291]
Hi guys!

First of all thank you for this svg library - it is just great!

A while ago, I forked this library with the goal to get it to run on android.
Meanwhile, we are using [SkiaSharp ](https://github.com/mono/SkiaSharp) for rendering.
This allows us to render on Android, iOS, Mac, Windows (and UWP) so basically everything except Linux.
Also, Skia supports hardware-rendering (OpenGL and Angle to translate the OpenGL calls to DirectX).

However, we did not just rip all the "System.Drawing" stuff out and replaced it, but refactored it into a factory with our own interfaces, so that you can (literally) stuff in any other render library you like.

My question is: Are you basically interested in merging this in? 
As we changed a lot (e.g. removed all TypeDescriptor stuff as this is not supported on UWP) so merging will be quite tedious.
On the other hand, I see a lot of benefits for you guys too.

In that case, please reach out to me.
Our repository is: [https://github.com/gentledepp/SVG](https://github.com/gentledepp/SVG)
The latest branch with plattform support is [https://github.com/gentledepp/SVG/tree/uwp_ios ](https://github.com/gentledepp/SVG/tree/uwp_ios )

Cheers!

**Comment by tebjan** [comment]
hello, this sounds totally interesting and actually we are also planning to use SkiaSharp internally in other projects. so that would fit quite well.

i'll discuss it with my colleagues and get back to you.

**Comment by gentledepp** [comment]
Great - the build is actually already working and our tests run on ios, android, uwp and windows.
Maybe we could arrange a skype chat sometime. (I am based in Austria, ... also sollten wir kein Zeitzonenproblem haben ;-))

**Comment by tebjan** [comment]
we had a short chat about the merge and came to the conclusion, that it would be better for us to schedule the merge bit further. even as far as july/august since we are preparing for a big event.
the merge could take a week or more of work. also a GDI implementation or at least forms adapter is needed...

**Comment by gentledepp** [comment]
sure - I am not in a hurry at all. 
My skype nickname is the same as my github name.
Feel free to drop me a line ;-)

**Comment by tebjan** [comment]
hello @gentledepp how is the status of the fork on your end? do you still maintain it? 
maybe its more easy to release a final version of this master branch and then completely switch to yours. some PRs might get lost, but overall your branch looks like the was to go.

**Comment by gentledepp** [comment]
Hi!
Yes, it is still mainained.
However, we are only supporting a "PCL" version.
There are no plans for porting to .NET Core at the moment

**Comment by tebjan** [comment]
great, thanks for the update!

**Comment by gentledepp** [comment]
What are your plans now?
We are happy to help lifting this codebase to a new level 👍

**Comment by mrbean-bremen** [comment]
@tebjan - can you elaborate about your plans? If you plan to switch to / integrate with the skiasharp branch, this may have change the priorities of the other issues, or make them obsolete.

**Comment by tebjan** [comment]
ideally we would merge the current master with the skiasharp master and combine all fixes. however, this sounds like a huge job and i am not sure whether we have enough time for that. 
i could also imagine to get the latest PRs regarding netstandard into master, releasing a few more fixes for the nuget 2 and prepare a version 3 which is based on skia. 
however, if @mrbean-bremen and @gentledepp are up for the task of merging, i'll do my best in order to support it.

**Comment by mrbean-bremen** [comment]
This looks like a huge task, as the code bases seem to have deviated sufficiently.
Before starting this, we have to know where we want to go - e.g. what about .NET compatibility (which is not available in the skiasharp branch, as far as I understand, but in the long run will go away anyway).
Also, I read that PCL is deprecated in favor of netstandard, so we may want to go to netstandard directly. As has been discussed in another issue, netstandard compatibility for .NET Core seems not to be far away, so maybe this is something we can wait for while we sort out a merge strategy. Increasing the test coverage for different platform is also something that has to be considered IMHO. 

Unfortunately, I myself have basically no experience in the related technologies - I'm only using .NET, and only under Windows so far (and only occassionally, most of the time using C++/Qt and Python), so while I'm willing to help, I'm hoping for you (@tebjan) and @gentledepp (and whoever is working on skiasharp now) to lead the work and make the decisions.
